File: src/main/java/org/shark/example/controller/QuartzController.java
package org.shark.example.controller;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.shark.example.service.base.pojo.ResponseDto;
import org.shark.example.service.quartz.CreateJobService;
import org.shark.example.service.quartz.DeleteJobService;
import org.shark.example.service.quartz.FindJobService;
import org.shark.example.service.quartz.pojo.CreateJobDto;
import org.shark.example.service.quartz.pojo.JobDto;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RequiredArgsConstructor
@RestController
@RequestMapping("/quartz")
@Slf4j
public class QuartzController {

    private final CreateJobService createJobService;
    @PostMapping
    public ResponseDto<Void> createJob(@RequestBody CreateJobDto createJobDto) {
        return createJobService.start(createJobDto);
    }

    private final DeleteJobService deleteJobService;

    @DeleteMapping
    public ResponseDto<Void> deleteJob(@RequestParam("jobName") String jobName, @RequestParam("jobGroup") String jobGroup) {
        return deleteJobService.start(jobName, jobGroup);
    }

    private final FindJobService findJobService;

    @GetMapping
    public ResponseDto<List<JobDto>> findJob() {
        return findJobService.start();
    }
}

File: src/main/kotlin/org/shark/example/controller/quartz/QuartzController.kt
package org.shark.example.controller.quartz

import org.shark.example.service.base.pojo.ResponseDto
import org.shark.example.service.quartz.CreateJobService
import org.shark.example.service.quartz.DeleteJobService
import org.shark.example.service.quartz.FindJobService
import org.shark.example.service.quartz.pojo.CreateJobDto
import org.shark.example.service.quartz.pojo.JobDto
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/quartz")
class QuartzController(
    private val createJobService: CreateJobService,
    private val deleteJobService: DeleteJobService,
    private val findJobService: FindJobService
) {
    private val log = LoggerFactory.getLogger(QuartzController::class.java)

    @PostMapping
    fun createJob(@RequestBody createJobDto: CreateJobDto): ResponseDto<Void> {
        return createJobService.start(createJobDto)
    }

    @DeleteMapping
    fun deleteJob(@RequestParam("jobName") jobName: String, @RequestParam("jobGroup") jobGroup: String): ResponseDto<Void> {
        return deleteJobService.start(jobName, jobGroup)
    }

    @GetMapping
    fun findJob(): ResponseDto<List<JobDto>> {
        return findJobService.start()
    }
}
